File: express/models/orderProduct.js
const { Model, DataTypes } = require("sequelize");
const sequelize = require("../config/database");

class OrderProduct extends Model {
  static associate(models) {
    OrderProduct.belongsTo(models.Order, {
      foreignKey: "order_id",
    });
    OrderProduct.belongsTo(models.Product, {
      foreignKey: "product_id",
    });
  }
}

OrderProduct.init(
  {
    order_id: {
      type: DataTypes.BIGINT,
      allowNull: false,
      primaryKey: true,
      references: {
        model: "Order",
        key: "id",
      },
    },
    product_id: {
      type: DataTypes.BIGINT,
      allowNull: false,
      primaryKey: true,
      references: {
        model: "Product",
        key: "id",
      },
    },
    quantity: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 1,
    },
  },
  {
    sequelize,
    modelName: "OrderProduct",
    tableName: "order_products",
    timestamps: false,
  }
);

module.exports = OrderProduct;

File: asp/performance_asp/performance_asp/Models/OrderProduct.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace performance_asp.Models
{
    public class OrderProduct
    {
        [ForeignKey("Order")]
        [Column("order_id")]
        public int OrderId { get; set; }

        [ForeignKey("Product")]
        [Column("product_id")]
        public int ProductId { get; set; }

        [Required]
        [Column("quantity")]
        public int Quantity { get; set; }

        public virtual Order Order { get; set; }
        public virtual Product Product { get; set; }
    }
}
